index.bs

```diff
@@ -1215,6 +1215,383 @@ implementation of this step is specific to the user agent.", the controlling
+  codec-parameters: [* codec-parameter]
+  time-scale: uint ; AKA clock rate; probably 90khz
+  ? default-duration: uint ; AKA ptime
+  ? default-rotation: video-rotation
+}
+
+data-encoding-offer = {
+  encoding-id: uint
+  format: text ; "text-cue" defined to mean text cues
+  format-parameters: [* codec-parameter] ; format-specific
+  time-scale: uint ; AKA clock rate; probably 1mhz
+}
+
+audio-stream-request = {
+  encoding-id: uint
+}
+
+video-stream-request = {
+  encoding-id: uint
+  ? target-resolution: video-resolution ; displayed resolution; try to be slightly above; default is unknown
+  ? max-framerate: float
+}
+
+data-stream-request = {
+  encoding-id: uint
+}
+
+-->
+
+
+Audio {#streaming-audio}
+------------------------------
+
+Senders may send audio to receivers by sending audio-frame messages (see
+[[#appendix-a]]) with the following keys and values.  An audio frame message
+contains a set of encoded audio samples for a range of time. A series of
+encoded audio frames that share a codec, codec parameters and a timeline form an
+audio encoding.
+
+Unlike most Open Screen Protocol messages, this one uses an
+array-based grouping rather than a struct-based grouping.  For
+required fields, this allows for a more efficient use of bytes on the
+wire, which is important for streaming audio because the payload is
+typically so small and every byte of overhead is relatively large.  In
+order to accomodate optional values in the array-based grouping, one
+optional field in the array is used to hold all optional values in a
+struct-based grouping.  This will hopefully provide a good balance of
+efficiency and flexibility.
+
+To further increase efficiency, each audio-frame message must be sent
+in a separate QUIC stream without the length prefix.  It must only use
+the type key prefix and then the encoded CBOR message immediately
+after.  Separate QUIC streams also allow audio frames to be received
+out of order.
+
+: encoding-id
+:: Identifies the media encoding to which this audio frame belongs.  This can be
+    used to reference properties of the encoding such as the codec, codec
+    properties, time scale (aka clock rate or sample rate), and default
+    duration.  Referencing properties of the encoding through the encoding id
+    helps to avoid sending duplicate information in every frame.
+
+: start-time
+:: Identifies the beginning of the time range of the audio frame.  The time
+    scale is inferred from the properties of the encoding.  The end time can be
+    inferred from the start time and duration.
+
+: duration
+:: If present, the duration of the audio frame.  The time
+    scale is inferred from the properties of the encoding.  Likewise, if not
+    present, the duration is inferred from the properties of the encoding.
+
+: sync-time
+:: If present, a time used to synchronize the start time of this audio frame (and
+    thus, this encoding) with that of other media encodings on different
+    timelines.
```

Comment:
It can be wall clock, but doesn't need to be.  It just for comparing different timelines, and so is arbitrary.

It could be used to sync in the same session and across sessions, potentially.